mlflow/gemini/autolog.py

```diff
@@ -49,18 +56,78 @@ def patched_class_call(original, self, *args, **kwargs):
 
 
 def patched_module_call(original, *args, **kwargs):
+    """
+    This method is used for patching standalone functions of the google.generativeai module.
+    This patch creates a span and set input and output of the original function to the span.
+    """
     config = AutoLoggingConfig.init(flavor_name=mlflow.gemini.FLAVOR_NAME)
 
     if config.log_traces:
         with mlflow.start_span(
             name=f"{original.__name__}",
             span_type=_get_span_type(original.__name__),
         ) as span:
-            inputs = construct_full_inputs(original, *args, **kwargs)
+            inputs = _construct_full_inputs(original, *args, **kwargs)
             span.set_inputs(inputs)
             result = original(*args, **kwargs)
             # need to convert the response of generate_content for better visualization
             outputs = result.to_dict() if hasattr(result, "to_dict") else result
             span.set_outputs(outputs)
 
             return result
+
+
+def _parse_outputs(outputs) -> list[ChatMessage]:
+    """
+    This method extract chat messages from genai.types.generation_types.GenerateContentResponse
+    """
+    # content always exist on output
+    # https://github.com/googleapis/googleapis/blob/9e966149c59f47f6305d66c98e2a9e7d9c26a2eb/google/ai/generativelanguage/v1beta/generative_service.proto#L490
+    return sum(
+        [
+            parse_gemini_content_to_mlflow_chat_messages(candidate.content)
+            for candidate in outputs.candidates
+        ],
+        [],
+    )
+
+
+def _log_tool_definition(model, span):
+    # when tools are not passed or object is ChatSession
```

Comment:
```suggestion
    # when tools are not passed
```
nit: since we check `isinstance(self, genai.GenerativeModel)` above (or maybe we can remove that check)